### ISSUE TITLE
Construction Paste Copying Block

### ISSUE THREAD

**Issue by Ronaldi2001** [Direwolf20-MC/BuildingGadgets#471]
Playing in my 1.12.2 server and using 2.8.3 of Building Gadgets. I was using the copy and paste gadgetzz zzz, when pasting the server crashed and when the server went back up all the actual block were placed instead of the paste, when I broke the blocks I got the real blocks. This crash also took all extra paste out of my inventory.

**Comment by MajorTuvok** [comment]
Could you provide the logs?
And also please reformat you message according to the template. This is very annoying. (Ideally you'd also list your Problems instead of putting them into a Text, but there are mutliple people who don't do that, so we'll live)

**Comment by PeritusDux** [comment]
* Forge version
14.23.5.2847
* Building Gadgets version
2.8.2
* Not a crash, but logs
https://paste.gg/p/anonymous/ac2c78e4f77545d7840997d6da93b260
* How to reproduce the issue
Paste https://paste.gg/p/anonymous/ffa9d3e3aa584641bcaff060dd96550c
* A screenshot, if applicable
![2020-05-02_12 44 05](https://user-images.githubusercontent.com/16677095/80872158-dde42480-8c75-11ea-8e67-105f36d3ac8a.png)
![2020-05-02_12 44 01](https://user-images.githubusercontent.com/16677095/80872161-df155180-8c75-11ea-8622-21ec110aba15.png)

Hi, we're also having this issue. We are using modpack McEternal 1.3.5. The server would time everyone out on the server and would place some real blocks down. I'm not sure if more recent versions of the mod fix this issue, but I figured it was worth a report since some people are still on 1.12.2.

**Comment by MajorTuvok** [comment]
I'm still not sure I understand the issue...
You place blocks and don't have the resources, so it uses construction paste. But then the Server times out?

**Comment by MajorTuvok** [comment]
On a side note: paste.gg is unusable on a mobile - however I could see that some mod with mushroom in the package name seems to be responsible for the logged error...
Can't really look it through though, without actually looking at it from my laptop.

**Comment by MichaelHillcox** [comment]
So effectively you think it's a dupe bug? are you able to get the crash log?

**Comment by MajorTuvok** [comment]
> So effectively you think it's a dupe bug?
Err, no... I don't understand the issue - I thought it was about the Server timing out.

**Comment by MajorTuvok** [comment]
`java.io.UTFDataFormatException: encoded string too long: 181239 bytes`
As well as notifications that the Server is running behind, but nothing which I could directly relate to BuildingGadgets...

**Comment by PeritusDux** [comment]
It is placing the real blocks instead of construction paste blocks. In those screenshots the build I copied had piles of dyed wool, blocks of emeralds, lapis, nether bricks, etc. When I pasted the build in our survival server it had construction blocks along with real blocks even though we didn't have any end stone for example. The stack of lapis blocks in the inventory was completely from construction paste.

**Comment by MichaelHillcox** [comment]
Hmm, sounds like a me issue, intresting, This only happen with the copy paste tool?

**Comment by PeritusDux** [comment]
I have only tested it with the copy paste tool. It also seems to only work when the server times me out. When there are less blocks involved it seems to place correctly.

**Comment by PeritusDux** [comment]
For context, I sliced a massive build from creative mode into sections by copying as many blocks as the tool would allow and then systematically pasting those maxed out sections into our survival server through the clipboard process. I bound the tool to a barrel from YABBA full of stone to cut down on construction paste use. I originally had the stone in a /dev/null, but switched hoping it would stop the time outs. It occurred in both instances.

**Comment by MichaelHillcox** [comment]
The time outs are because you're pasting a max size build on a server that likely can't handle it, the server should reduce the build size allowed :P I plan on added a low lag option to the config to reduce the impact of large builds but I don't plan on bringing it back to 1.12.

**Comment by PeritusDux** [comment]
Works for me. It's not a big issue we were just surprised to find the real blocks where we expected construction blocks. Figured it was worth a heads up. It's amazing we were able to copy the build between worlds it worked fantastically, only a little too well it seems.  I can't speak for Ronaldi2001, but we just deleted the real blocks to try and retain the survival legitimacy. Thanks for taking the time to check this out.
